### ISSUE TITLE
Dynamic component Router null

### ISSUE THREAD

**Issue by dmitweb** [zeit/next.js#7677]
# Bug report

## Describe the bug

In dynamic component Router null and initialize only after loading

## To Reproduce

1. npm run dev

## Expected behavior

Initialized Router with asPath, pathname, query

## Additional context

Git repo https://github.com/dmitweb/next.js-dynamic_import_router

**Comment by ijjk** [comment]
Hi, this is expected. We recommend using `withRouter` to access the router instance inside of your components. In the repo you linked you are trying to access the `Router` before it has been fully initialized. Using `withRouter` prevents this. 

I'm going to close this, if you're still having trouble with this feel free to reply with additional info.

**Comment by dmitweb** [comment]
Hi @ijjk. I know what use withRouter give me Router instance when React is rendering. Can u please show usage withRouter at which Router instance be initialized in loader, or loading. I find only way to do this, move line:
https://github.com/zeit/next.js/blob/canary/packages/next/client/index.js#L151
here:
https://github.com/zeit/next.js/blob/canary/packages/next/client/index.js#L167
Router instance now available on client, but i don't know maybe it will break something...
I need Router instance in dynamic component loading on client side for my task. Maybe u show me alternative method do that.
ps. i don't need router in react component render, only in loading for ex. load diff chunks on client

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
